hs_core/tasks.py

```diff
@@ -455,6 +459,13 @@ def create_bag_by_irods(resource_id):
             # gets deleted by another request when being downloaded
             istorage.runBagitRule(bagit_rule_file, bagit_input_path, bagit_input_resource)
             istorage.zipup(irods_bagit_input_path, bag_path)
+            if res.raccess.published:
+                # compute checksum to meet DataONE distribution requirement
+                chksum = istorage.checksum(bag_path)
+                extra_data = res.extra_data
+                extra_data['bag_checksum'] = chksum
+                res.extra_data = extra_data
+                res.save()
```

Comment:
I think this logic belongs in the resource class.

Reply:
Can you clarify a bit on your thought? It simply set a bag_checksum extra_data field with no extra logic. What is the benefit to move it to the resource class?

Reply:
Just helps keep things clean and organized.  I think a resource should know how to save and access a bag_checksum.  An outside process shouldn't need to know that bag_checksum is a key in extra_data.

Reply:
OK. Makes sense to me now. Will move it to the resource class as a property.